File: src/main/java/org/fletchly/genius/di/ContextModule.java
package org.fletchly.genius.di;

import dagger.Module;
import dagger.Provides;
import org.bukkit.plugin.java.JavaPlugin;
import org.fletchly.genius.context.data.DatabaseManager;
import org.fletchly.genius.context.data.DatabaseManagerImpl;
import org.fletchly.genius.util.ConfigurationManager;

import javax.inject.Singleton;

@Module
public class ContextModule {
    @Provides
    @Singleton
    DatabaseManager provideDatabaseManager(JavaPlugin javaPlugin, ConfigurationManager configurationManager) {
        return new DatabaseManagerImpl(javaPlugin, configurationManager);
    }
}

File: src/main/kotlin/io/fletchly/genius/context/ContextModule.kt
package io.fletchly.genius.context

import dagger.Module
import dagger.Provides
import io.fletchly.genius.config.ConfigManager
import io.fletchly.genius.context.service.ConcurrentHashMapContextService
import io.fletchly.genius.context.service.ContextService
import javax.inject.Singleton

@Module
class ContextModule {
    @Provides
    @Singleton
    fun provideContextService(configManager: ConfigManager): ContextService = ConcurrentHashMapContextService(configManager)
}